packages/broker/src/config/ConfigWizard.ts

```diff
@@ -1,254 +1,539 @@
-import { toEthereumAddress } from '@streamr/utils'
+import { ConfigFile, getDefaultFile } from './config'
 
-export interface PrivateKeyAnswers extends Answers {
-    generateOrImportPrivateKey: 'Import' | 'Generate'
-    importPrivateKey?: string
-}
+const MinBalance = utils.parseEther('0.1')
+
+export const start = async (): Promise<void> => {
+    log(`
+        >
+        > ***Welcome to the Streamr Network!***
+        > This Config Wizard will help you setup your node.
+        >
+        > The steps are documented here:
+        > *https://docs.streamr.network/guides/how-to-run-streamr-node#config-wizard*
+        >
+    `)
+
+    try {
+        const privateKey = await getPrivateKey()
+
+        const nodeAddress = new Wallet(privateKey).address
+
+        const network = await getNetwork()
+
+        const operator = await getOperatorAddress()
+
+        const operatorPlugins = operator ? {
+            operator: {
+                operatorContractAddress: operator,
+            },
+        } : {}
+
+        const { http, ...pubsubPlugins } = await getPubsubPlugins()
+
+        if (http) {
+            Object.assign(pubsubPlugins, {
+                http: {},
+            })
+        }
+
+        const storagePath = await getStoragePath()
+
+        const httpServer = http?.port ? { port: http.port } : void 0
+
+        const config: ConfigFile = {
+            client: {
+                auth: {
+                    privateKey,
+                },
+            },
+            plugins: {
+                ...operatorPlugins,
+                ...pubsubPlugins,
+            },
+            httpServer,
+        }
+
+        persistConfig(
+            storagePath,
+            network === 'polygon' ? config : getMumbaiConfig(config)
+        )
+
+        log(`
+            >
+            > ~ *Congratulations, you've setup your Streamr node!*
+            > Your node address is ${chalk.greenBright(nodeAddress)}
+            > Your node's generated name is ${chalk.greenBright(getNodeMnemonic(privateKey))}
+        `)
+
+        if (operator) {
+            const resume = progress((f) =>
+                style(`> Your node address has *${f} MATIC* _– checking balance…_`)
+            )
+
+            try {
+                const balance = await getNativeBalance(network, nodeAddress)
+
+                const content = `Your node address has *${utils
+                    .formatEther(balance)
+                    .replace(/\.(\d+)/, (f) => f.substring(0, 3))} MATIC*`
+
+                resume()
+
+                if (balance.lt(MinBalance)) {
+                    log(`
+                        > ! ${content}. You'll need to fund it with a small amount of MATIC tokens.
+                    `)
+                } else {
+                    log(`> ${content}`)
+                }
+            } catch (e) {
+                resume()
+
+                log('> x Failed to fetch node\'s balance')
+            }
+        }
+
+        if (operator) {
```

Comment:
Coudl use explicit `undefined` checks here and in other if statements

Reply:
`operator` is either `undefined` or a valid ethereum address.

Reply:
It already covers the case and is sufficiently obvious. I'll leave it.